homeassistant/components/octoprint/binary_sensor.py

```diff
@@ -22,14 +22,15 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     devices = []
     for octo_type in monitored_conditions:
+        metadata = BINARY_SENSOR_TYPES[octo_type]
         new_sensor = OctoPrintBinarySensor(
             octoprint_api,
             octo_type,
-            BINARY_SENSOR_TYPES[octo_type][2],
+            metadata.key,
             name,
-            BINARY_SENSOR_TYPES[octo_type][3],
-            BINARY_SENSOR_TYPES[octo_type][0],
-            BINARY_SENSOR_TYPES[octo_type][1],
+            metadata.unit,
+            metadata.endpoint,
+            metadata.group,
```

Comment:
That is a bit cumbersome. We do have a `NamedTuple` so why pass all items individually? It's enough to split them in `__init__` like the rest.

Not sure about `metadata = BINARY_SENSOR_TYPES[octo_type]`. For all other PRs we set it in `__init__`, but doing it here might make more sense.

Reply:
Yes that is one of many optimizations we can make, but I prefer to make the intervention as small as possible. Especially now if these PRs are to be used to test out another direction.